cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -239,10 +241,10 @@ func MustNew(cfg config.LocalConfig, endpoint string, adminEndpoint string) *Dae
 		ReadTimeout: defaultReadTimeout,
 	}
 	if adminEndpoint != "" {
-		adminMux := http.NewServeMux()
+		adminMux := supporthttp.NewMux(logger)
+		// after importing net/http/pprof, debug endpoints are implicitly registered in the default serve mux
+		adminMux.Handle("/", http.DefaultServeMux)
 		adminMux.Handle("/metrics", promhttp.HandlerFor(d.prometheusRegistry, promhttp.HandlerOpts{}))
-		adminMux.HandleFunc("/debug/pprof/heap", pprof.Index)
-		adminMux.HandleFunc("/debug/pprof/profile", pprof.Profile)
```

Comment:
@tamirms FYI, we were missing some pprof endpoints, see https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/net/http/pprof/pprof.go;l=85-91